### ISSUE TITLE
Safari Extension think it's in Private mode after every browser restart

### ISSUE THREAD

**Issue by Vsubb** [bitwarden/browser#2297]
### Steps To Reproduce

Set Safari to work in standard (not private) mode

1. Start Safari with some pages opened
2. Click on Bitwarden Safari extension button in browser toolbar.
3. Select Login (small empty window opens under button)
4. Click again on Bitwarden Safari extension button in browser toolbar.
5. Message appears under button, that BW window not available in browser's private mode
![2022-01-28_15-15-54](https://user-images.githubusercontent.com/7082356/151557368-c54b614a-0cf1-4364-af8a-24afe6a82336.png)

Once again - Safari is not in private mode.

### Expected Result

Login window

### Actual Result

Error message about private mode (browser really is not in private mode)

### Screenshots or Videos

![2022-01-28_15-15-54](https://user-images.githubusercontent.com/7082356/151557368-c54b614a-0cf1-4364-af8a-24afe6a82336.png)

### Operating System

macOS

### Operating System Version

Monterey 12.1

### Web Browser

Safari

### Browser Version

15.2 (17612.3.6.1.6)

### Build Version

1.55.0

**Comment by reznet** [comment]
I see this as well in Safari 15.3.  I literally just upgraded my 2015 MacBook Pro from Catalina to Monterey, and the first time I tried to open the Bitwarden extension, I got this error as well.  My Bitwarden application version is 1.30.0, which is currently the latest in the App Store.

![Screen Shot 2022-01-29 at 3 48 33 PM](https://user-images.githubusercontent.com/2987293/151681400-923bcee1-d1c2-49b5-a7f4-7825a26752f5.png)

**Comment by reznet** [comment]
An update on my experience - the plugin is now working correctly.  I don't know why, but I'll explain what I did and why.  After upgrading to Monterey, safari and chrome pages would regularly crash.  On YouTube, for example, the tab would crash if I right clicked a link.  And most tabs crashed even if they sat there for a few seconds.  After I unplugged my PreSonus Firestudio Mobile Firewire audio interface, connected via the apple firewire to thunderbolt 2 adapter, the problem went away.  I could surf the web happily and listen to audio out the MacBook internal speakers.  I ended up uninstalling the PreSonus drivers, rebooting, and the crashing tabs problem went away.  I then re-enabled the Bitwarden extension and it worked.  So it may be that a problem in safari caused by an incompatible audio driver was causing Bitwarden to think it could not load.  I hope this update helps anyone else with the same issue.

**Comment by eliykat** [comment]
The test we use for Private Mode is whether the popup can communicate with the extension's background page. This bug indicates that the popup has fallen out of sync with the rest of the extension. Can you try restarting, or disabling and re-enabling the extension, and see if the problem persists?

**Comment by pepper77** [comment]
I had the exact same problem on macOS 12.2 and Safari 15.2. I tried disabling and re-enabling extension as suggested, and everything worked as expected again. Thanks.

**Comment by Vsubb** [comment]
What I tried to check:
1. Disable BitWarden extension in Safari Settings, then re-enable it without browser restart - it asked me master-password ant then works nice till browser restart/
2. After browser restart problem reappears - "Unfortunately this window is not available in private mode for this browser."

**Comment by pepper77** [comment]
You are right. Now I have updated to MacOS 12.2.1 and Safari 15.3, and after a restart of the Safari-browser, the problem reappears. Only difference is that with new versions of MacOS and Safari I don't see the above mentioned text, it is just an empty window with a spinning "pause animation" that appears. After disabling and re-enabling the extension once more, things are working again (until the next restart of the browser).

**Comment by ctschach** [comment]
> You are right. Now I have updated to MacOS 12.2.1 and Safari 15.3, and after a restart of the Safari-browser, the problem reappears. Only difference is that with new versions of MacOS and Safari I don't see the above mentioned text, it is just an empty window with a spinning "pause animation" that appears. After disabling and re-enabling the extension once more, things are working again (until the next restart of the browser).

Similar behaviour here - no way to access the extension in Safari. When I click on it, I see a while windows with the spinning thing in the middle. And even if I wait for 5 minutes it doesn't go away. Disabling and re-enabling doesn't help. I also deinstalled bitwarden complete with all files in ~/Library and reinstalled it from the app-store. Still no success.

**Comment by lubbertkramer** [comment]
Any update on this issue, Safari keeps showing that the private mode is on

Safari: Version 15.5 (17613.2.7.1.8)
MacOS: Montery 12.4 (21F79)

**Comment by ejl371** [comment]
It is seemingly random when this happens.

**Comment by turnipjs** [comment]
> ![Screenshot 2022-07-03 at 00 11 36](https://user-images.githubusercontent.com/8224586/177017535-41cccb5d-6b7b-49a8-86a1-81ba239134dd.png)

I get this same message with:
Safari: Version 16.2 (18614.3.7.1.5)
MacOS: Ventura Version 13.1 (22C65)
BitWarden: 2022.11.0

**Comment by caewok** [comment]
Definitely still happening. Possibly related, it seems really slow when opening the browser extension in Safari.
Safari: 17.5 (19618.2.12.11.6)
MacOS: 14.5 (23F79)
Bitwarden: 2024.4.1

**Comment by closebot-bw** [comment]
⚠️ **Stale Issue Notice**

This issue has been automatically marked as stale due to inactivity. It will be closed in **2 weeks** (October 15, 2025) if no further activity occurs.

If this issue is still relevant and you would like to keep it open, please:
- Comment on this issue to show continued interest
- Provide any additional information or updates
- Confirm that the issue still exists in the latest version

Thank you for your contribution to this project! 🙏

**Comment by closebot-bw** [comment]
🔒 **Issue Closed Due to Inactivity**

This issue has been automatically closed due to lack of activity for an extended period. We periodically review and close inactive issues to help maintain our issue tracker and focus on current priorities.

**If this issue is still relevant:**
- Please create a new issue with updated information
- Include steps to reproduce the problem if it's a bug report
- Mention if this issue still occurs in the latest version

Thank you for your contribution to this project. Your feedback helps us improve! 🙏
